db/migrations/20210427142108_migrate_stripe_subscription.js

```diff
@@ -0,0 +1,100 @@
+require('dotenv').config()
+
+const Stripe = require('stripe')
+
+const t_subscription = 'circle_subscription'
+
+const t_subscription_item = 'circle_subscription_item'
+
+exports.up = async (knex) => {
+  const secret = process.env['MATTERS_STRIPE_SECRET']
+
+  if (!secret) {
+    console.error('Stripe secret is not provided')
+    return
+  }
+
+  const stripeAPI = new Stripe(secret, { apiVersion: '2020-03-02' })
+
+  // fetch migrated subscription items
+  const items = await knex
+    .select(
+      'circle_subscription_item.*',
+      'circle_subscription.provider_subscription_id'
+    )
+    .from(t_subscription_item)
+    .join(
+      t_subscription,
+      'circle_subscription_item.subscription_id',
+      'circle_subscription.id'
+    )
+    .where({
+      'circle_subscription_item.archived': true,
+      'circle_subscription_item.provider': 'stripe',
+      'circle_subscription_item.remark': 'trial_migration',
+    })
+
+  const total = (items || []).length
+
+  // process subscription items
+  for (const [index, item] of items.entries()) {
+    try {
+      console.log('-------------------------------')
+      console.log(`Process ${index + 1}/${total} item: ${item.id}`)
+
+      // fetch stripe subscription item
+      const stripeItems = await stripeAPI.subscriptionItems.list({
+        subscription: item.provider_subscription_id,
+      })
+
+      if (!stripeItems || !stripeItems.data || stripeItems.data.length <= 0) {
+        console.log('Subscription items not found')
+        continue
+      }
+
+      // fetch stripe subscription
+      const stripeSubscription = await stripeAPI.subscriptions.retrieve(
+        item.provider_subscription_id
+      )
+
+      if (!stripeSubscription) {
+        console.log('Stripe subscription not found')
+        continue
+      }
+
+      if (stripeSubscription.canceled_at > 0) {
+        console.log('Stripe subscription arleady canceled')
+        continue
+      }
```

Comment:
also need to check if subscription items already been deleted?

Reply:
Nice point. Added.